pages/implementation/devs-101.mdx

```diff
@@ -6,18 +6,134 @@ title: 'Avo 101 for developers'
+<Link passHref href="/implementation/avo-functions-overview"><a>Avo Functions</a></Link> are 
+type safe analytics wrappers that are code generated based on your <Link passHref href="/workspace"><a>tracking plan</a></Link>. <br/> <br/> 
 
-## Avo Inspector SDK
+Technical or non-technical people can define analytics event schemas in the <Link passHref href="/workspace"><a>Avo Tracking Plan</a></Link>, 
+and you can generate human readable type safe code based on that. The days of unstructured 
+and outdated tracking plan spreadsheets and Jira tickets are over 🥳 So are the days of brittle 
+analytics implementation you have to revisit to fix again and again and again 😌
+
+### How do Avo Functions work?
+
+Avo Functions are type safe analytics wrappers that can work with any analytics tool you already 
+use or plan on using, whether it's any of the third-party event logging SDKs (e.g. 
+Google Analytics 4, Segment, Rudderstack, mParticle, Snowplow, Amplitude, Mixpanel, Pendo, 
+etc) or your home made API.
+
+The Avo Functions have a <Link passHref href="/implementation/destinations"><a>Destination Interface</a></Link> 
+that you have full control over how you implement. You provide a simple Destination Interface 
+object when initializing Avo Functions, so that any time an Avo Function calls an analytics 
+logging method (e.g. logEvent, setUserProperties, etc), you can govern exactly what analytics 
+SDK method should be called.
+
+In the <Link passHref href="/workspace"><a>Avo Tracking Plan</a></Link> you can configure which analytics actions 
+(for example you might want to update user properties or log revenue along with logging an event)
+should be triggered with an Avo Function, and the Avo Functions 
+will handle calling all the relevant Destination Interface methods. All you have to do is call the Avo Function in your 
+code base and pass in type safe key-value pairs, and the Avo Function will take care of the rest.
+
+Learn more technical details about <Link passHref href="/implementation/avo-functions-overview"><a>Avo Functions here</a></Link>.
+
+### Do events flow through Avo servers?
+
+No. None of your events ever go through Avo servers. They go directly from your devices (whether that's 
+a front-end client or a server) to the analytics destination.
+
+This is compatible to Segment's "device mode". With Segment, you can choose 
+between <Link passHref href="https://segment.com/docs/guides/intro-impl/#connection-modes"><a>_Cloud-mode_ or _Device-mode_</a></Link>:
+- In _Cloud-mode_, your events go through Segment servers. Your devices send the events to the 
+Segment cloud (aka the Segment Servers) and the Segment cloud sends them to the analytics destinations.
+- In _Device-mode_ the events go directly from your devices (whether it's a front end client 
+or a server) to the analytics destination.
+
+In this regard, Avo works like Segment's _Device-mode_.
```

Comment:
I'm wondering if this is too much detail on Segment's cloud vs device mode. My main concern is that people will somehow read this as Segment (and Segment's device-mode) being a required tool for using Avo.

Reply:
Maybe if we move it into 

> this thingy

it will make more sense?

Reply:
Interesting. Right. I think this is helpful context, but we can tweak it to focus less on Segment and call it CDPs.

Reply:
Updated.